Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2_test.py

```diff
@@ -1077,3 +1138,145 @@ def test_do_soap_request(self, mocker, requests_mock, integration_context, is_ne
+                '<Value>2024-12-11T11:11:24.433385Z</Value>'
+                '</DateComparisonFilterCondition>',
+                id='Date greater than condition',
+            ),
+            pytest.param(
+                FilterConditionTypes.text,
+                'Contains',
+                'Incident Priority',
+                '456',
+                'High',
+                # Expected
+                '<TextFilterCondition>'
+                '<Operator>Contains</Operator>'
+                '<Field name="Incident Priority">456</Field>'
+                '<Value>High</Value>'
+                '</TextFilterCondition>',
+                id='Text contains condition',
+            ),
+        ]
+    )
+    def test_construct_generic_filter_condition(
+        self,
+        condition_type: FilterConditionTypes,
+        operator: str,
+        field_name: str,
+        field_id: str,
+        search_value: str,
+        expected_xml_condition: str
+    ):
+        """
+        Given:
+            - Case 1: A date filter condition with 'GreaterThan' operator on the 'Last Updated' field.
+            - Case 2: A text filter condition with 'Contains' operator on the 'Incident Priority' field.
+        When:
+            - Calling construct_generic_filter_condition.
+        Assert:
+            - Case 1: Ensure a valid DateComparisonFilterCondition XML element with the correct sub-elements.
+            - Case 2: Ensure a valid TextFilterCondition XML element with the correct sub-elements.
+        """
+        xml_condition = construct_generic_filter_condition(
+            condition_type=condition_type,
+            operator=operator,
+            field_name=field_name,
+            field_id=field_id,
+            search_value=search_value,
+        )
+        assert xml_condition == expected_xml_condition
+
+    @pytest.mark.parametrize(
+        'logical_operator, conditions_count, expected_operator_logic',
+        [
+            pytest.param(
+                'or',
+                3,
+                # Expected
+                '<OperatorLogic>1 OR 2 OR 3</OperatorLogic>',
+                id='OR with 3 conditions',
+            ),
+            pytest.param(
+                'AND',
+                4,
+                # Expected
+                '<OperatorLogic>1 AND 2 AND 3 AND 4</OperatorLogic>',
+                id='AND with 4 conditions',
+            ),
+            pytest.param(
+                'XOR',
+                1,
+                # Expected
+                '',
+                id='XOR with 1 condition',
+            ),
+        ]
+    )
+    def test_construct_operator_logic(self, logical_operator: str, conditions_count: int, expected_operator_logic: str):
+        """
+        Given:
+            - Case 1: 'or' logical operator and 3 conditions.
+            - Case 2: 'AND' logical operator and 4 conditions.
+            - Case 3: 'XOR' logical operator with 1 condition.
```

Comment:
no need to elaborate on each case
```suggestion
            - arguments for constructing an operator logic
```

Reply:
Fixed in commit cbeb616e474a7c5d67a4e15ac24b85286fcb58c8